FindRubric endpoint

a5x55555 opened the pull request:
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039

a5x55555 committed:
FindRubric endpoint v1.0

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039

MariyaSin commented on src/WikiService.Business/Commands/Rubric/FindRubricCommand.cs:
```diff
@@ -0,0 +1,54 @@
+  {
+    private readonly IBaseFindFilterValidator _baseFindValidator;
+    private readonly IRubricRepository _rubricRepository;
+    private readonly IFindRubricMapper _findRubricMapper;
```
the mapper name must point to the model returned by this mapper

MariyaSin commented on src/WikiService.Business/Commands/Rubric/FindRubricCommand.cs:
```diff
@@ -0,0 +1,54 @@
+    public FindRubricCommand(
+      IBaseFindFilterValidator baseFindValidator,
+      IRubricRepository rubricRepository,
+      IFindRubricMapper findRubricMapper,
```
same

MariyaSin commented on src/WikiService.Business/Commands/Rubric/FindRubricCommand.cs:
```diff
@@ -0,0 +1,54 @@
+
+      (List<DbRubric> dbRubrics, int totalCount) = await _rubricRepository.FindAsync(filter);
+
+      response.Body = dbRubrics?.Select(dbRubric => _findRubricMapper.Map(dbRubric)).ToList();
```
https://www.youtube.com/watch?v=5Lit18RJYp8

MariyaSin commented on src/WikiService.Data/RubricRepository.cs:
```diff
@@ -1,16 +1,66 @@
 using LT.DigitalOffice.WikiService.Data.Provider;
+using LT.DigitalOffice.WikiService.Models.Db;
+using LT.DigitalOffice.WikiService.Models.Dto.Requests.Filters;
+
```
remove empty lines

MariyaSin commented on src/WikiService.Data/RubricRepository.cs:
```diff
@@ -1,16 +1,66 @@
+      {
+        dbRubric = dbRubric.Where(
+          rubric =>
+            rubric.Name.Contains(filter.NameIncludeSubstring));
```
maybe it's better to find in ignore case?

MariyaSin commented on src/WikiService.Mappers/Db/FindRubricMapper.cs:
```diff
@@ -0,0 +1,26 @@
+
+namespace LT.DigitalOffice.WikiService.Mappers.Db
+{
+  public class FindRubricMapper : IFindRubricMapper
```
see command

MariyaSin commented on src/WikiService.Mappers/Db/FindRubricMapper.cs:
```diff
@@ -0,0 +1,26 @@
+        return null;
+      }
+
+      return new FindRubricRequest
```
it is not request data...
rename

MariyaSin commented on src/WikiService.Mappers/Db/FindRubricMapper.cs:
```diff
@@ -0,0 +1,26 @@
+        Id = dbRubric.Id,
+        Name = dbRubric.Name,
+        ParentId = dbRubric.ParentId,
+        ChildCount = dbRubric.ChildCount,
```
every time value is null here

MariyaSin commented on src/WikiService.Mappers/Db/Interfaces/IFindRubricMapper.cs:
```diff
@@ -0,0 +1,12 @@
+namespace LT.DigitalOffice.WikiService.Mappers.Db.Interfaces
+{
+  [AutoInject]
+  public interface IFindRubricMapper
```
rename

MariyaSin commented on src/WikiService.Models.Db/DbRubric.cs:
```diff
@@ -12,6 +12,7 @@ public class DbRubric
     public Guid Id { get; set; }
     public string Name { get; set; }
     public Guid? ParentId { get; set; }
+    public int? ChildCount { get; set; }
```
for what?

a5x55555 replied:
Just to tell front end side are there any child rubrics or not

MariyaSin commented on src/WikiService.Models.Dto/Requests/Filters/FindRubricFilter.cs:
```diff
@@ -0,0 +1,14 @@
+    public string NameIncludeSubstring { get; set; }
+
+    [FromQuery(Name = "isAscendingSort")]
+    public bool? IsAscendingSort { get; set; }
```
add sort by isActive status

MariyaSin commented on src/WikiService.Models.Dto/Requests/FindRubricRequest.cs:
```diff
@@ -0,0 +1,17 @@
+
+namespace LT.DigitalOffice.WikiService.Models.Dto.Requests
+{
+  public record FindRubricRequest
```
it is not request

MariyaSin commented on src/WikiService.Models.Dto/Requests/FindRubricRequest.cs:
```diff
@@ -0,0 +1,17 @@
+    public Guid Id { get; set; }
+    public string Name { get; set; }
+    public Guid? ParentId { get; set; }
+    public int? ChildCount { get; set; }
```
it can be easier

a5x55555 committed:
FindRubric endpoint v1.1

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039

a5x55555 committed:
FindRubric endpoint v1.2

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039

a5x55555 committed:
FindRubric endpoint v1.3

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039

VLiamin commented on src/WikiService.Data/RubricRepository.cs:
```diff
@@ -1,16 +1,82 @@
+          ? dbRubric.Where(rubric => rubric.IsActive == true)
+          : dbRubric.Where(rubric => rubric.IsActive == false);
+      }
+
```
May be remove line?

VLiamin commented on src/WikiService.Data/RubricRepository.cs:
```diff
@@ -1,16 +1,82 @@
+        dbRubric = dbRubric.OrderByDescending(rubric => rubric.CreatedAtUtc);
+      }
+
+      foreach(var rubric in dbRubric)
```
May be replace var? Because i do not know what type is it.

VLiamin commented on src/WikiService.Data/RubricRepository.cs:
```diff
@@ -1,16 +1,82 @@
+
+      foreach(var rubric in dbRubric)
+      {
+        foreach (var id in _provider.Rubrics.AsQueryable())
```
And here

a5x55555 committed:
FindRubric endpoint v1.4

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039

a5x55555 committed:
FindRubric endpoint v1.5

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039

a5x55555 committed:
FindRubric endpoint v1.6

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039

fixed folders, filenames and namespaces

MariyaSin commented on src/WikiService.Data/RubricRepository.cs:
```diff
@@ -1,16 +1,92 @@
+
+      if (filter.IsActive.HasValue)
+      {
+        dbRubric = filter.IsActive.Value
```
Where(rubric => rubric.IsActive == filter.IsActive.Value

MariyaSin commented on src/WikiService.Data/RubricRepository.cs:
```diff
@@ -1,16 +1,92 @@
+      }
+
+      return dbRubric;
+    }
```
add empty line

a5x55555 committed:
FindRubric endpoint v1.7

http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1039
